### ISSUE TITLE
AWS ECS on EC2: InvalidParameterException: Assign public IP is not supported for this launch type

### ISSUE THREAD

**Issue by nickatsegment** [hashicorp/waypoint#643]
**Describe the bug**
Creating a simple ECS on EC2 deployment always (?) results in an error: `InvalidParameterException: Assign public IP is not supported for this launch type`

```
    deploy {
        use "aws-ecs" {
          cluster = "nick-testing-waypoint"
          ec2_cluster = true
          memory = 512
          region = "us-west-2"
        }
    }
```

**Additional context**

Just a quick look [at the code](https://github.com/hashicorp/waypoint/blob/f302419072e36162d92f8cf6b4bc17e33bdc351f/builtin/aws/ecs/platform.go#L1039) shows `AssignPublicIp` getting set unconditionally to `ENABLED`. [The terraform docs for this setting](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ecs_service#assign_public_ip) say this is only for Fargate, which I guess is true, though [the AWS docs](https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_AwsVpcConfiguration.html) do not.

**Comment by nickatsegment** [comment]
Here's what I had to do to get this working:

1. hack out [the code](https://github.com/hashicorp/waypoint/blob/f302419072e36162d92f8cf6b4bc17e33bdc351f/builtin/aws/ecs/platform.go#L1039) that sets `AssignPublicIp: ENABLED` in the CLI
2. spin up a NAT gateway for my ECS cluster so it can access the server's LB

I guess giving every container a public IP is just so that it can reach the server? Hopefully it's on the roadmap to support NAT :)

Alternatively, I guess you could provision a second *internal* load balancer for the server, and advertise that. That LB has to be in the same VPC as your ECS cluster though, unless you want to deal with peering.

**Comment by evanphx** [comment]
The public ip is because of how Fargate gets access to the container registry. Without having a public IP, a nat gateway is required to access ECR even.

**Comment by nickatsegment** [comment]
Huh. I suppose that makes sense. Looks like you can also [create a VPC PrivateLink](https://docs.aws.amazon.com/AmazonECR/latest/userguide/vpc-endpoints.html) (I'm no Fargate pro).

**Comment by mitchellh** [comment]
Fixed by #724

**Comment by nickatsegment** [comment]
Thank you!
